forge-game/src/main/java/forge/game/ability/effects/DiscoverEffect.java

```diff
@@ -0,0 +1,178 @@
+package forge.game.ability.effects;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.Iterables;
+import forge.game.Game;
+import forge.game.ability.AbilityKey;
+import forge.game.ability.AbilityUtils;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.Card;
+import forge.game.card.CardCollection;
+import forge.game.card.CardLists;
+import forge.game.card.CardZoneTable;
+import forge.game.cost.CostDiscard;
+import forge.game.cost.CostPart;
+import forge.game.cost.CostReveal;
+import forge.game.player.Player;
+import forge.game.spellability.LandAbility;
+import forge.game.spellability.SpellAbility;
+import forge.game.trigger.TriggerType;
+import forge.game.zone.PlayerZone;
+import forge.game.zone.Zone;
+import forge.game.zone.ZoneType;
+import forge.util.CardTranslation;
+import forge.util.Localizer;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class DiscoverEffect extends SpellAbilityEffect {
+
+    @Override
+    protected String getStackDescription(SpellAbility sa) {
+
+        return sa.getActivatingPlayer() + " discovers " + sa.getParamOrDefault("Num", "1") + ".";
+    }
+
+    @Override
+    public void resolve(SpellAbility sa) {
+        final Card host = sa.getHostCard();
+        final Game game = host.getGame();
+        final Player p = getDefinedPlayersOrTargeted(sa).get(0);
```

Comment:
you could put this in a loop, so we could handle if multiple players should discover at the same time
(it that is ever needed)